Add documentation notes about legacy cluster type & data access (#2437)
databricks/terraform-provider-databricks: docs/resources/mount.md

### Example mounting ADLS Gen2 with AAD passthrough

> -> **Note** Please note that AAD passthrough is considered a legacy data access pattern, please use Unity Catalog for fine grained data access control.
> 
> -> **Note** You can't create AAD passthrough mount using service principal!

Reviewer: ```suggestion
-> **Note** AAD passthrough is considered a legacy data access pattern. Use Unity Catalog for fine-grained data access control.

-> **Note** Mounts using AAD passthrough cannot be created using a service principal.
```
